lib/lvol/lvol.c

```diff
@@ -1268,9 +1268,13 @@ create_lvol_snapshot(struct spdk_lvol *origlvol, const char *snapshot_name,
 	req->lvol = newlvol;
 	req->cb_fn = cb_fn;
 	req->cb_arg = cb_arg;
-
 	spdk_bs_create_snapshot(lvs->blobstore, spdk_blob_get_id(origblob), &xattr_args.lvol_xattrs,
 				lvol_create_cb, req);
+	/*
```

Comment:
This logic should be involved once snapshot is successfully created, before calling the completion callback 'lvol_create_cb'. Otherwise we could have snapshot creation error - reported via callback, and the cleared cache.

Reply:
fix now. created new callback for snapshot to hold parent lvol to reset cache before callback is called.